script.js

```diff
@@ -0,0 +1,120 @@
+// base class
+function Parent(value) { 
+    this.value = value;
+}
+
+Parent.prototype.checkout = function () {
+  if (Number.isInteger(this.value) == true) {
+    return "number";
+  } else {
+    return "string";
+  }
+};
+
+Parent.prototype.get = function () {
+  return this.value;
+};
+
+Parent.prototype.plus = function (...args) {
+  for (let element of args) {
+    this.value += element;
+  }
+  return this.value;
+};
+
+Parent.prototype.minus = function (...args) {
+  if (this.checkout() == "number") {
+    for (let element of args) {
+      this.value -= element;
+    }
+    return this.value;
+  } else if (this.checkout() == "string") {
+    this.value = this.value.slice(0, 0 - args[0]);
+    return this.value;
+  }
```

Comment:
what is the expected result in case the value doesn't belong to any of the types you've made check?